src/components/OAuthLoginButton.tsx

```diff
@@ -1,97 +1,55 @@
-import React, { FC, useCallback } from 'react';
-import {
-  Platform,
-  TouchableOpacity,
-  TouchableOpacityProps,
-  View,
-  ViewStyle,
-  Text,
-  TextStyle,
-} from 'react-native';
+import React, { useCallback } from 'react';
+import { TextStyle, TouchableOpacityProps, ViewStyle } from 'react-native';
 import { useStyles } from '../../src/hooks/useStyles';
 import { LoginParams, useOAuthFlow } from '../../src/hooks/useOAuthFlow';
-import { tID } from '../common/testID';
 import { Theme } from './BrandConfigProvider';
+import { Button } from 'react-native-paper';
 
-type OAuthLoginButtonParams = Omit<TouchableOpacityProps, 'onPress'> &
+type Props = Omit<TouchableOpacityProps, 'onPress'> &
   LoginParams & {
     label: string;
     styles?: OAuthLoginButtonStyles;
   };
 
-export const OAuthLoginButton: FC<OAuthLoginButtonParams> = ({
+export function OAuthLoginButton({
   onSuccess,
   onFail,
   label,
   styles: instanceStyles,
-  ...touchableOpacityProps
-}) => {
+  ...buttonProps
+}: Props) {
   const { styles } = useStyles(
     'OAuthLoginButton',
     defaultStyles,
     instanceStyles,
   );
   const { login } = useOAuthFlow();
 
-  const _login = useCallback(async () => {
+  const handleOnPress = useCallback(async () => {
     await login({
       onSuccess,
       onFail,
     });
   }, [login, onSuccess, onFail]);
 
   return (
-    <View style={styles.button}>
-      <TouchableOpacity
-        testID={tID('oauth-login-button')}
-        {...touchableOpacityProps}
-        onPress={_login}
-        style={styles.touchableOpacity}
-      >
-        <View style={styles.content}>
-          <Text style={styles.label}>{label}</Text>
-        </View>
-      </TouchableOpacity>
-    </View>
+    <Button
+      mode="contained"
+      onPress={handleOnPress}
+      {...styles}
+      {...buttonProps}
+    >
+      {label}
+    </Button>
```

Comment:
All of Paper's Button [props](https://callstack.github.io/react-native-paper/docs/components/Button/) like `buttonColor`, `textColor`, `loading`, `icon`, `disabled`, etc are still available on an instance level.